### ISSUE TITLE
random_reference to nicknames

### ISSUE THREAD

**Issue by prescod** [SFDO-Tooling/Snowfakery#261]
This could be implemented through a database of all generated records, with a nickname column or joined table. This implementation would also have the benefit of allowing the retrieval of data from random_reference'd records.

The cost is that every outputstream would need storage.

Also: restoring loaded datatypes for data could be a challenge.

**Comment by mitchwd** [comment]
Our team has a need for this too. For a semi real-world example:
We're currently using the Account object to hold two distinct types of data: Clinics and Schools.
When generating a Contact, we need to set the School the Contact attends, and the Clinic they were referred by...

Ideally, we would be able to do something like this:

```
- object: Account
  nickname: school
  count: 3
  fields:
    BillingCity: 
      fake: City
    Name: ${{BillingCity}} High School
    # ...more school-specific data

- object: Account
  nickname: clinic
  count: 5
  fields:
    BillingCity: 
      fake: City
    Name: ${{BillingCity}} Clinic
    # ...more clinic-specific data

- object: Contact
  nickname: client
  count: 10
  fields:
    Name:
      fake: Name
    School: 
      random_reference: school
    Referrer: 
      random_reference: clinic
```

In the meantime, is there any known workaround?

**Comment by prescod** [comment]
I am actually working on this feature today. If your counts are really as small as 3 and 5 then you can do this as a workaround in the meantime:

```
- macro: SchoolStuff
  fields:
    BillingCity:
      fake: City
    Name: ${{BillingCity}} High School
    # ...more school-specific data

- object: Account
  nickname: school1
  include: SchoolStuff
- object: Account
  nickname: school2
  include: SchoolStuff
- object: Account
  nickname: school3
  include: SchoolStuff

- macro: ClinicStuff
  fields:
    BillingCity:
      fake: City
    Name: ${{BillingCity}} Clinic
    # ...more clinic-specific data

- object: Account
  nickname: clinic1
  include: ClinicStuff
- object: Account
  nickname: clinic2
  include: ClinicStuff
- object: Account
  nickname: clinic3
  include: ClinicStuff
- object: Account
  nickname: clinic4
  include: ClinicStuff
- object: Account
  nickname: clinic5
  include: ClinicStuff

- object: Contact
  nickname: client
  count: 10
  fields:
    Name:
      fake: Name
    School:
      random_choice:
        - random_reference: school1
        - random_reference: school2
        - random_reference: school3
    Referrer:
      random_choice:
        - random_reference: clinic1
        - random_reference: clinic2
        - random_reference: clinic3
        - random_reference: clinic4
        - random_reference: clinic5
```

**Comment by prescod** [comment]
Fixed in Snowfakery 3.2

https://github.com/SFDO-Tooling/Snowfakery/releases/tag/v3.2
